hr_employee_service_contract/models/hr_employee.py

```diff
@@ -42,22 +42,31 @@ def _compute_first_contract_id(self):
             )
 
     @api.multi
-    @api.depends('contract_ids', 'contract_ids.date_end')
+    @api.depends('contract_ids', 'contract_ids.date_end', 'contract_ids.state')
     def _compute_last_contract_id(self):
         Contract = self.env['hr.contract']
         for employee in self:
-            employee.last_contract_id = Contract.search(
-                employee._get_contract_filter(),
-                order='date_end desc',
-                limit=1
-            )
-
+            employee.last_contract_id = self._get_last_contract_id(employee.id)
+    
     @api.multi
     @api.onchange('service_hire_date')
     def _onchange_service_hire_date(self):  # pragma: no cover
         # Do nothing
         pass
 
+    @api.multi
+    def _get_last_contract_id(self, employee_id):
```

Comment:
@YuriyGural if you're changing the behaviour, please describe the change as well as provide tests for it

Reply:
```
    def test_6(self):
        employee = self.SudoEmployee.create({
            'name': 'Employee #6',
        })
        contract_1 = self.Contract.create({
            'name': 'Employee #6 Contract #1',
            'employee_id': employee.id,
            'wage': 5000.0,
            'state': 'open',
            'date_start': self.today - relativedelta(years=5),
            'date_end': self.today - relativedelta(years=1),
        })
        contract_2 = self.Contract.create({
            'name': 'Employee #6 Contract #1',
            'employee_id': employee.id,
            'wage': 6000.0,
            'state': 'draft',
            'date_start': self.today - relativedelta(years=2),
            'date_end': False,
        })

        self.assertEqual(
            employee.service_termination_date,
            self.today - relativedelta(years=1)
        )

        contract_2.write({'state': 'open'})
        self.assertEqual(
            employee.service_termination_date,
            False
        )
```

Reply:
https://github.com/OCA/hr/blob/12.0/hr_employee_service/tests/test_hr_employee_service.py